### DISCUSSION TITLE
Deep Learning

### DISCUSSION THREAD

**Discussion by rafif24** [teamalgoritma/community#462] [Q&A] [answered]
Halo jadi saya mempunyai banyak pertanyaan yang bisa diliat disini 
[2440057741.docx](https://github.com/teamalgoritma/community/files/10069357/2440057741.docx)

dari soal - soal yang ada disitu saya masih belum mengerti karena kebetulan saya mengikuti theia batch yang dimana deep learning itu masih minggu depan dan saya mempunyai tugas ini yang dimana deadline nya hari jumat.
Saya minta bantuan untuk menjelaskan step step yang harus dilakukan (dalam bentuk notebook kalau bisa) dan apabila memang lama mungkin bisa memberikan link referensi untuk bahan pembelajaran

**notes** : bahasa yang digunakan python

**Comment by VictorNugraha** [comment]
Dear @rafif24,

Terima kasih atas pertanyaan yang disampaikan, pertanyaan Anda sudah kami terima. Kami akan mencoba membantu mencarikan refrensi jawaban untuk pertanyaan yang diberikan.

**Comment by yosiaazarya** [comment, marked as answer]
Dear @rafif24,

Berikut adalah jawaban yang dapat kami berikan terkait permasalahan yang ada. Sebelumnya kami tidak bisa menjawab tugas yang berikan secara keseluruhan sehingga kami akan memberikan referensi sebagai guidance untuk mengerjakan tugas tersebut. Dataset yang tercantum di soal juga tidak dapat dibuka akibat keterbatasan akses, sehingga referensi yang akan kami berikan adalahyang bersifat umum terkait dengan pengerjaan dari metode yang diperintahkan.

Untuk case di nomor 1 adalah sebagai berikut:
1. (opsi 1)Berikut adalah referensi untuk membuat NN  secara manual -> https://machinelearningmastery.com/implement-backpropagation-algorithm-scratch-python/
2. (opsi 2)Berikut adalah referensi untuk membuat NN dengan keras -> https://machinelearningmastery.com/tutorial-first-neural-network-python-keras/ ; disarankan dengan opsi 2
3. Adapun untuk case yang diberikan adalah bersifat klasifikasi, sehingga jika ingin mencari referensi lebih lanjut, pastikan menggunakan keyword seperti "classification neural network python" dsb
4. Pastikan melakukan CROSS VALIDATION (poin b di soal no. 1) sesuai dengan keterangan yang diberikan
5. Diperlukan adanya 2 hidden layer yang berisikan 2 x n  node yang artinya adalah 2 kali dari jumlah prediktor dan output layer sejumlah kelas target yang akan diprediksi sehingga arsitektur yang dimiliki adalah sebagai berikut: 
MISAL jumlah prediktor = 5
input layer (jumlah node=5) -> hidden layer 1 (jumlah node=10) -> hidden layer 2 (jumlah node=10) -> output layer (jumlah node=2)
6. Setelah terbentuk arrsitektur, gunakan arsitektur tersebut untuk training data, kemudian melakukan pengecekkan akurasi dan melakukan tuning (googling keyword : "keras NN tuning python")

Untuk case nomor 2 adalah sebagai berikut:
1. Referensi untuk resize gambar menjadi resolusi tertentu : https://towardsdatascience.com/how-to-resize-images-using-python-8aaba74602ed
2. Referensi untuk membuat Convolutional Neural Network dengan keras : https://www.analyticsvidhya.com/blog/2021/08/beginners-guide-to-convolutional-neural-network-with-implementation-in-python/
3. Pastikan melakukan CROSS VALIDATION sesuai dengan ketentuan yang diberikan
4. Pastikan untuk membuat arsitektur sesuai dengan gambar yang ditentukan di soal (jangan membuat arsitektur sesuai referensi)

Demikian adalah jawaban yang dapat kami berikan, semoga dapat membantu menyelesaikan permasalahan yang ada. Good luck!

**Reply by rafif24** [reply]
oke terima kasih banyak ya ka

**Reply by rafiff23** [reply]
ka untuk nomor 2 dari link referensi agak berbeda dengan dataset saya ya, kalo bisa di cek kan itu data_train image didalem subfolder kalo test tidak. Cara load data keduanya gimana ya? dan untuk split data train jadi x_train dan y_train itu gimana?
